pyeudiw/openid4vp/presentation_submission/__init__.py

```diff
@@ -0,0 +1,148 @@
+import re
+from pydantic import ValidationError
+import importlib
+from typing import Any
+import logging
+
+from pyeudiw.openid4vp.presentation_submission.base_vp_parser import BaseVPParser
+from pyeudiw.openid4vp.presentation_submission.schemas import PresentationSubmissionSchema
+from pyeudiw.openid4vp.presentation_submission.exceptions import (
+    MissingHandler, 
+    MalformedPath, 
+    SubmissionValidationError, 
+    VPTokenDescriptorMapMismatch,
+    ParseError
+)
+
+logger = logging.getLogger(__name__)
+
+class PresentationSubmissionHandler:
+    def __init__(self, formats: list[dict], max_submission_size: int = 4096) -> None:
+        """
+        Initialize the PresentationSubmissionHandler handler with the submission data.
+
+        :param submission: The presentation submission data.
+        :type submission: dict[str, Any]
+        :param config: Configuration dictionary.
+        :type config: dict[str, Any], optional
+        :param args: Additional arguments to be passed to handlers.
+        :type args: Any
+        :param kwargs: Additional keyword arguments to be passed to handlers.
+        :type kwargs: Any
+
+        :raises KeyError: If the 'format' key is missing in the submission.
+        :raises ValueError: If the format is not supported or not defined in the configuration.
+        :raises ImportError: If the module or class cannot be loaded.
+        :raises ValidationError: If the submission data is invalid or exceeds size limits.
+        """
+        if not formats:
+            raise ValueError("At least one format must be defined.")
+
+        self.max_submission_size = max_submission_size
+        self.handlers: dict[str, BaseVPParser] = {}
+
+        for format_conf in formats:
```

Comment:
this loader should be used also for duckle query language, therefore I would not use it only within presentation exchange. this loader is about the satosa backend and its configuration, not the presentation exchange alone